app/client/websocket/index.ts

```diff
@@ -339,15 +367,28 @@ export default class WebSocketClient {
         this.connectFailCount = 0;
         this.responseSequence = 1;
         clearTimeout(this.connectionTimeout);
+        clearInterval(this.pingInterval);
         this.conn?.close();
     }
 
     public invalidate() {
         clearTimeout(this.connectionTimeout);
+        clearInterval(this.pingInterval);
         this.conn?.invalidate();
         this.conn = undefined;
     }
 
+    private ping() {
+        const msg = {
+            action: WebsocketEvents.PING,
+            seq: this.responseSequence++,
+        };
+
+        if (this.conn && this.conn.readyState === WebSocketReadyState.OPEN) {
+            this.conn.send(JSON.stringify(msg));
```

Comment:
is every server capable of responding with pong? we need to make sure this is backwards compatible with at least the latest ESR, if this causes no problems, then we can leave it like this.

Reply:
I believe ping/pong handling has been available since the beginning:

https://github.com/mattermost/mattermost/blob/ad343a0f4ad175053f7d0da12a0587bcbb396d1c/api/general.go#L76-L79

It looks like support was added more than 9 years ago 🙂